org.ekstep.reviewercomments-1.0/editor/plugin.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Plugin to get comments from portal
+ * @class reviewerComments
+ * @constructor
+ * @extends org.ekstep.contenteditor.basePlugin
+ * @author Revathi P 
+ * @listens stage:select
+ */
+
+org.ekstep.contenteditor.basePlugin.extend({
+    isApiCalled: false,
+    initialize: function () {
+        ecEditor.addEventListener("stage:select", this.initShowComments, this);
+        this.initShowComments();
+    },
+    initShowComments: function (event) {
+        var instance = this;
+        this.callAPI(function () {
+            setTimeout(function () {
+                instance.initializeComments();
+            }, 1000);
+        });
+    },
+    callAPI: function (callback) {
+        var instance = this;
+        if (!instance.isApiCalled) {
+            this.context = org.ekstep.contenteditor.globalContext;
+            if (!ecEditor._.isUndefined(this.context)) {
+                if (!ecEditor._.isUndefined(this.context.contentId) && this.context.contentId != "") {
+                    var instance = this;
+                    var data = {
+                    };
+                    //Do Api call and get the response  
+                    ecEditor.getService(ServiceConstants.CONTENT_SERVICE).getComments(data, function (error, response) {
+                        if (!error) {
+                            this.items = [];
+                            instance.response = response;
+                            //Loop through the response and generate the comments Thread
+                            instance.comments = instance.response.data.result.comments
+                            instance.isApiCalled = true;
+                            callback();
+                        }
+                    });
+                }
+            }
+        } else {
+            callback();
+        }
+    },
+
+    initializeComments: function () {
+        // console.log('event:', event)
+        var instance = this;
+        var items = [];
+        if (instance.comments !== undefined) {
+            //Display the comments by filtering the stage id from the instance.comments
+            var filtered_comments = _.filter(instance.comments, ['stageId', ecEditor.getCurrentStage().id]);
+
+            sortedComments = _.sortBy(filtered_comments, function (dateObj) {
+                return new Date(dateObj.createdOn);
+            });
+            if (filtered_comments.length == 0) {
+                jQuery('#reviewerComments').html('<div>No review comments</div>');
```

Comment:
Dont use jQuery, use editorApi method to access Jquery object.

Reply:
All string should define in single const object & use it where ever it is required.